internal/services/machinelearning/machine_learning_registry_data_source.go

```diff
@@ -0,0 +1,221 @@
+
+		"public_network_access_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"main_region": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"location": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"storage_account_type": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"hns_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Computed: true,
+					},
+
+					"system_created_storage_account_id": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"system_created_container_registry_id": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+
+		"replication_region": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"location": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"storage_account_type": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"hns_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Computed: true,
+					},
+
+					"system_created_storage_account_id": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"system_created_container_registry_id": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+
+		"identity": commonschema.SystemAssignedUserAssignedIdentityComputed(),
+
+		"discovery_url": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"intellectual_property_publisher": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"ml_flow_registry_uri": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"managed_resource_group": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"tags": commonschema.TagsDataSource(),
+	}
+}
+
+func (d MachineLearningRegistryDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MachineLearning.RegistryManagement
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model MachineLearningRegistryDataSourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding Machine Learning Registry data source model: %+v", err)
+			}
+
+			id := registrymanagement.NewRegistryID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			resp, err := client.RegistriesGet(ctx, id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("%s was not found", id)
+				}
+				return fmt.Errorf("reading %s: %+v", id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("reading nil model %s", id)
+			}
+
```

Comment:
Set the id here, and nil-check the model before processing data into state as per the [contributor guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/81ce454a4d57ef01c5189d2fadbc6530326770a7/contributing/topics/guide-new-data-source.md?plain=1#L220-L242)